### PR TITLE
[PoC] Progressbar component implementation

### PR DESCRIPTION
**Pull request by vladbailescu**
Progressbar component implementation

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/ProgressBar.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import org.osgi.annotation.versioning.ConsumerType;
+
+import com.adobe.cq.export.json.ComponentExporter;
+
+@ConsumerType
+public interface ProgressBar extends Component {
+
+    /**
+     * Name of property that defines the completion percentage.
+     * The property should provide a numeric value between 0 and 100.
```
Comment (joerghoh) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/ProgressBar.java:29]: "...numeric (**non-float**) value...."

File: content/src/content/jcr_root/apps/core/wcm/components/progressbar/v1/progressbar/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                                <items jcr:primaryType="nt:unstructured">
+                                    <column
+                                        jcr:primaryType="nt:unstructured"
+                                        granite:class="cq-RichText-FixedColumn-column"
+                                        sling:resourceType="granite/ui/components/coral/foundation/container">
+                                        <items jcr:primaryType="nt:unstructured">
+                                            <completed
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="granite/ui/components/coral/foundation/form/numberfield"
+                                                fieldDescription="The completion percentage, as a number between 0 and 100."
```
Comment (joerghoh) [inline content/src/content/jcr_root/apps/core/wcm/components/progressbar/v1/progressbar/_cq_dialog/.content.xml:53]: "as a **integer** number between" (to point out that it cannot be a float)

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/ProgressBar.java (hunk 1 above)
Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/ProgressBar.java:29]: I was wondering if we should support fractional numbers too.

**Comment by vladbailescu** [comment]
Updated the PR, added a Library example and some styling:

image

File: content/src/content/jcr_root/apps/core/wcm/components/progressbar/v1/progressbar/.content.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    cq:icon="progressbar"
+    jcr:description="Displays a progress bar indicator."
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/progressbar/v1/progressbar/.content.xml:4]: I'd describe it as: "Displays a progress bar."

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/components/progressbar/.content.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2019 Adobe
```
Comment (jckautzmann) [inline examples/ui.apps/src/content/jcr_root/apps/core-components-examples/components/progressbar/.content.xml:3]: Should be 2020.

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/components/progressbar/clientlibs/.content.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2019 Adobe
```
Comment (jckautzmann) [inline examples/ui.apps/src/content/jcr_root/apps/core-components-examples/components/progressbar/clientlibs/.content.xml:3]: Should be 2020.

Comment (vladbailescu) [reply examples/ui.apps/src/content/jcr_root/apps/core-components-examples/components/progressbar/clientlibs/.content.xml:3]: 🤦

File: content/src/content/jcr_root/apps/core/wcm/components/progressbar/v1/progressbar/progressbar.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+<div data-sly-use.progressBar="com.adobe.cq.wcm.core.components.models.ProgressBar"
+     id="${progressBar.id}"
+     class="cmp-progress">
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/progressbar/v1/progressbar/progressbar.html:18]: I'd call the class `cmp-progressbar` and all sub-classes `cmp-progressbar__*` (even `cmp-progressbar__bar`)

**Comment by vladbailescu** [comment]
@jckautzmann , implemented your feedback in bc332ab and 06bc146

**Pull request merged**